### PR TITLE
Removes Hugger hardstun and replaces it with a new carrier ability

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

The hardstun just does a lot of stamina damage and a tiny aesthetic knockdown now

The new ability uses huggers to fling back marines from a carrier after a small channel by the carrier, and the range scales with the amount of huggers the carrier was storing at the time

## Why It's Good For The Game

Hardstuns are bad and New abilities is good

## Changelog
:cl:
balance: Removed hugger hardstun
add: Replaced the hugger stun with a new carrier ability powered by huggers: psychic wave
imageadd: New action sprite by Kuro
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm [hunk 1]
```diff
@@ -215,3 +215,53 @@
...
+		if(!silent)
+			to_chat(X, "<span class='xenowarning'>We don't have any young ones!</span>")
+		return FALSE
+
+/datum/action/xeno_action/psychic_wave/action_activate()
+	var/mob/living/carbon/xenomorph/carrier/X = owner
+
+	if(!do_after(X, 0.5 SECONDS, FALSE, X, BUSY_ICON_DANGER))
+		return fail_activate()
+	var/huggers = length(X.huggers)
```
Comment (Hughgent) [inline code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm:249]:
```suggestion
	var/total_huggers = length(X.huggers)
```

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm [hunk 2]
```diff
@@ -215,3 +215,53 @@
...
+	X.visible_message("<span class='warning'>[X] crushes the huggers on it's back, releasing a burst of energy!</span>",\
+		"<span class='xenowarning'>We contract our spines, crushing [huggers] little ones and releasing their psychic energy!</span>")
+	new /obj/effect/overlay/temp/emp_pulse(X.loc)
+	new /obj/effect/spawner/gibspawner/xeno(X.loc)
+	for(var/mob/living/victim in view(3))//yes this throws back friendly xenos too
+		if(victim == X)
+			continue
+		var/target = victim.loc
+		for(var/i=1 to huggers)
```
Comment (Hughgent) [inline code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm:258]:
```suggestion
		for(var/i=1 to total_huggers)
```

**Review by Hughgent — CHANGES_REQUESTED** [review]
alternative variable name `your_babies`

**Comment by Surrealaser** [comment]
Narratively this is one weird ass ability.

So you charge the facehugger like a bomb of psychic energy? What?

**Comment by Isy232** [comment]
Find some way to fit this in maybe a little more.

Like you kill your own children to power your psychic energy?

Dafuq?

**Comment by TiviPlus** [comment]
If you have better ideas for lore then suggest them but this works fine

File: code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm [hunk 3]
```diff
@@ -215,3 +215,53 @@
...
+		return FALSE
+
+/datum/action/xeno_action/psychic_wave/action_activate()
+	var/mob/living/carbon/xenomorph/carrier/X = owner
+
+	if(!do_after(X, 0.5 SECONDS, FALSE, X, BUSY_ICON_DANGER))
+		return fail_activate()
+	var/total_huggers = length(X.huggers)
+	X.visible_message("<span class='warning'>[X] crushes the huggers on it's back, releasing a burst of energy!</span>",\
+		"<span class='xenowarning'>We contract our spines, crushing [huggers] little ones and releasing their psychic energy!</span>")
```
Comment (Hughgent) [inline code/modules/mob/living/carbon/xenomorph/castes/carrier/abilities_carrier.dm:251]:
```suggestion
		"<span class='xenowarning'>We contract our spines, crushing [total_huggers] little ones and releasing their psychic energy!</span>")
```

**Review by Hughgent — APPROVED** [review]

**Comment by Surrealaser** [comment]
@TiviPlus Honestly I would just get rid of the insta-hug on direct hits and replace them with a stamina loss + stagger and slow; maybe with knockback while on harm intent.

**Comment by Hughgent** [comment]
image conflicts!

**Pull request closed without merging**